lib/filtrex/utils/encoder.ex

```diff
@@ -20,17 +20,33 @@ defmodule Filtrex.Utils.Encoder do
   """
   defmacro encoder(comparator, reverse_comparator, expression, values_function \\ {:&, [], [[{:&, [], [1]}]]}) do
     quote do
+      import Filtrex.Utils.Encoder
+
       def encode(condition = %{comparator: unquote(comparator), inverse: true}) do
         condition |> struct(inverse: false, comparator: unquote(reverse_comparator)) |> encode
       end
 
       def encode(%{column: column, comparator: unquote(comparator), value: value}) do
+        values = 
+          unquote(values_function).(value)
+          |> intersperse_column_refs(column)
+
         %Filtrex.Fragment{
-          expression: String.replace(unquote(expression), "column", column),
-          values: unquote(values_function).(value)
+          expression: String.replace(unquote(expression), "column", "?"),
+          values: values
         }
       end
     end
   end
 
+  def intersperse_column_refs(values, column) do
```

Comment:
This function seems reasonable to me but it should either have a `docstring`, be private, or be embedded into the macro `quote` block.

Reply:
I'll give it a `@doc` tag. I didn't put it into the macro `quote` block because it doesn't use anything from the generated module's context. When a functional solution works, I tend to use it instead of macros.

Reply:
@danielberkompas Agreed. I have the same perspective about embedding it. I think actual Jose mentioned it specifically at some point as a recommendation.